Report accounts

lomamech opened the pull request:
Report Accounts
- Implementation of client side 
- Implementation of API and Report PDF Design
- Implementation of integration test ans E2E test

@sfount  @mbayopanda

lomamech committed:
Implementation reportAccounts

lomamech committed:
Improve report of accounts

lomamech committed:
Resolve conflict with Branch Master

lomamech committed:
Complete report Accounts and improve the design of PDF

lomamech committed:
Test integration for report Accounts

lomamech committed:
Test end 2 end for report Accounts

jniles commented on client/src/i18n/fr.json:
```diff
@@ -684,7 +684,7 @@
     "SELECT": {
       "EMPTY"              : "Vide",
-      "ACCOUNT"            : "Sélectionner compte",
+      "ACCOUNT"            : "Sélectionner un compte",
```
:+1:

jniles commented on client/src/partials/reports/modals/report_accounts.config.js:
```diff
@@ -0,0 +1,54 @@
+  vm.cancel = ModalInstance.dismiss;
+  vm.report = report;
+
+  Accounts.read(null, { detailed: 1, is_auxiliary: 0})
```
You don't need `is_auxiliary` here...

jniles commented on client/src/partials/reports/modals/report_accounts.config.js:
```diff
@@ -0,0 +1,54 @@
+    .catch(Notify.errorHandler);
+
+  function generate(form) {
+    var url = 'reports/finance/report_accounts';
```
This report is for a single account, correct?  I propose that this URL should be `reports/finance/account`.

jniles commented on client/src/partials/reports/modals/report_accounts.config.js:
```diff
@@ -0,0 +1,54 @@
+
+    var options = {
+      account_id: vm.account.id,
+      account_label: vm.account.label,
```
Do you actually need the parameters `account_label` and `account_number`?

lomamech replied:
Yes, For display in the PDF report

jniles replied:
Ah, I see it now.  :+1:

jniles commented on server/controllers/finance/reports/reportAccounts/report.handlebars:
```diff
@@ -0,0 +1,54 @@
+{{> head title="PATIENT_RECORDS.FINANCIAL_ACTIVITY.REPORT_TITLE" }}
```
I think this title is wrong.

jniles commented on server/controllers/finance/reports/reportAccounts/report.handlebars:
```diff
@@ -0,0 +1,54 @@
+    <table class="table table-condensed table-striped table-report table-bordered">
+      <thead>
+        <tr class="text-capitalize text-center" style="background-color: #ddd;">
+          <th style="width: 10%;">{{translate "TABLE.COLUMNS.INVOICE_ID" }}</th>
```
Only class 4 accounts will have invoice IDs.  I propose you remove this column for a generic account report.

jniles reviewed (changes requested):
@lomamech, the code is pretty good.  I've made a few comments to improve.

For functionality, I propose you:
1. Remove the `Invoice ID`, since most accounts will never have an invoice against them.  You should remove it from the SQL as well.
2. Add an option to include the posting journal that is `false` by default.  Currently, both the General Ledger and Posting Journal is shown by default.  It should be just the General Ledger by default, with an option to enable the Posting Journal.

lomamech committed:
Remove Invoice ID in report

lomamech committed:
Update interface for Rapport Accounts
Adding the way to select the source of Data
Update test integration and test end to end

lomamech committed:
Update url for the single account

lomamech committed:
Improve design for PDF reports
